### PR TITLE
Change root password

### PR DESCRIPTION
**Pull request by Sylvain303**
Try to fix it that way.

Need to be completed on non debian, non mariaDB.

### CODE REVIEW

**Comment by gravyboat** [comment]
Was this actually ready to be merged?

**Comment by Sylvain303** [comment]
Well, I've done some strange branch stuff, but the diff looks like quite good.

Sorry, I didn't realize the dependencies in my branches, and this formula is quite difficult to hack for a salt beginner.

As mentioned in #120, it is a *work around*, a *sysadmin sugar*,  that I commonly use with `.my.cnf`. This fix for changing root password becomes doable with the previous root password stored in `.my.cnf` on the minion. It works on my debian + mariaDB. I need help to test it on other platform.

It also lack *unchanged state*, as it always run. It also displays root password on the screen, and maybe in log, also mentioned in #120, and #43.

Not tested extensively (or not remembering) what happen if you disable `enable_root_my_cnf`

~~~yaml
mysql:
  server:
    enable_root_my_cnf: False
~~~

It should not break formula as I've done:

~~~diff
 mysql/init.sls
@@ -16,6 +16,9 @@ include:
 {% if mysql_dev %}
   - mysql.dev
 {% endif %}
+{% if salt['pillar.get']('mysql:server:enable_root_my_cnf', False) %}
+  - mysql.change_root_password
+{% endif %}
~~~

I would like to put it under unittesting but I lack time, and knowledge, hardware, about testing salt's formula.
Some help would be nice.

**Comment by amontalban** [comment]
Hey guys,

I'm really interested in this PR, do you need help to get it merged?

Thanks!

**Comment by Sylvain303** [comment]
hi @amontalban ,

Did you test it? On which OS ? + which DB server ?
I tested only MariaDB.

I have an voucher to produce some VM on OVH public cloud, I can setup some distrib to help for testing purpose.

Regards,
Sylvain.

**Comment by amontalban** [comment]
Hi @Sylvain303,

Not yet, but will give it a try and let you know my results.

Thanks!

**Comment by Sylvain303** [comment]
I did not pull recently it may have some difference, See #120 for details and security concern.

File: mysql/files/root-my.cnf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{#-
+vim: set ft=jinja:
+
+This file is used for giving password less root access to shell's root user
+for command line tools: mysql, mysqladmin, mysqldump, etc.
+
+It is used with slat minion too in: /etc/salt/minion.d/
```
Comment (cboltz) [inline mysql/files/root-my.cnf]: Typo: slat -> salt

**Comment by esoel** [comment]
This pull request looks very interesting. I'll test on centos/mariadb as soon as I have time.

**Comment by noelmcloughlin** [comment]
Hi @Sylvain303 could you fix conflicts?  @esoel did you get time to test PR?   It would be good if #104, #106, and #120 could finally be progressed.

**Comment by noelmcloughlin** [comment]
Where is stalebot?  Closing.

**Pull request closed without merging**